python/ray/serve/_private/application_state.py

```diff
@@ -601,20 +601,26 @@ def _determine_app_status(self) -> Tuple[ApplicationStatus, str]:
         # application status to set.
         deployment_statuses = self.get_deployments_statuses()
         lowest_rank_status = min(deployment_statuses, key=lambda info: info.rank)
-        if lowest_rank_status.status == DeploymentStatus.UNHEALTHY:
+        if lowest_rank_status.status == DeploymentStatus.DEPLOY_FAILED:
+            failed_deployments = [
+                s.name
+                for s in deployment_statuses
+                if s.status == DeploymentStatus.DEPLOY_FAILED
+            ]
```

Comment:
Non-blocker: can probably refactor this loop into a helper and reuse for both here and the unhealthy case :)